src/mgmt/srv_drpc.c

```diff
@@ -289,6 +289,43 @@ create_pool_props(daos_prop_t **out_prop, char *owner, char *owner_grp,
 	return rc;
 }
 
+/* Returns either an error or the length of the comps array. */
+static int
+proto_fault_domains_to_pool_comps(Mgmt__FaultDomain **proto, size_t proto_len,
+				  struct pool_component **comps)
+{
+	struct pool_component	*tmp_comps;
+	size_t			n_comps = 0;
+	size_t			i;
+
+	if (proto == NULL || proto_len == 0) {
+		*comps = NULL;
+		return 0;
+	}
+
+	D_ALLOC_ARRAY(tmp_comps, proto_len);
+	if (tmp_comps == NULL)
+		return -DER_NOMEM;
+
+	/* Don't include the root node for the moment */
```

Comment:
I don't actually know why. :( A description or at least some hints on where to look at would be welcome.

Reply:
The way the pool map code works right now, it expects us to pass just the list of domains, which doesn't include the root. The pool map code creates the root node itself. I can clarify the comment.